concepts/functions/about.md

```diff
@@ -19,6 +22,55 @@ func PrintHelloName(name string) {
 }
 ```
 
+Parameters of the same type can be declared together, followed by a single type declaration.
+
+```go
+import "fmt"
+
+func PrintGreetingName(greeting, name string) {
+  fmt.Println(greeting + " " + name)
+}
+```
+
+## Parameters vs. Arguments
+
+Let's quickly cover two terms that are often confused together: `parameters` and `arguments`. 
+Function parameters are the names defined in the function's signature, such as `greeting` and `name` in the function `PrintGreetingName` above.
+Function arguments are the concrete values passed to the function parameters when we invoke the function.
+For instance, in the example below, `"Hello"` and `"Katrina"` are the arguments passed to the `greeting` and `name` parameters:
+
+```go
+PrintGreetingName("Hello", "Katrina")
+```
+
+## Pass by Value & Pass by Pointer
+
+It is also important to clarify the concept of passing by value as well as passing by pointer (sometimes referred to as _by reference_).
+In the example below, `name`, the `string` parameter to the function `PrintHelloName`, is passed into the function by value.
+This means that Go essentially makes a copy of the original value so that only this copy is accessed or modified by our function.
+
+```go
+import "fmt"
+
+func PrintHelloName(name string) {
+  fmt.Println("Hello " + name)
+}
+```
+
+Strictly speaking, all arguments are passed by value in Go, i.e. a copy is made of the value or data provided to the function.
+But if you don't want to make a copy of the data that you choose to pass to a function and
+want to actually access and change the data in the function,
+then you should use pointers as arguments, a.k.a. pass by reference.
+
+In Go, not only can we pass primitive values into a function, but we can also pass a `pointer` of these values
+(a reference to the data) to the function.
+This means that the function receives only a copy to the pointer instead of the underlying data that this pointer refers to.
+Note that we say _a copy to the pointer_ since everything is passed by value to functions in Go,
+but a pointer copy, in contrast to a value copy, gives us an "address" that points to the data we want to access and modify.
```

Comment:
While correct, I don't think going into this level of detail of trying to explain why a pointer is in reality just a value too is helpful here. Might just confuse the students even more. I like the simple explanation of the "Pointers" section, and we probably should just leave as that.